### ISSUE TITLE
Variable name is not displayed  in PMML file created

### ISSUE THREAD

**Issue by archanatholar05** [jpmml/sklearn2pmml#309]
I ran below code where first I have created sklearn  pipeline and done fit. Later I have used PMML libraries  to create PMML file. In the final PMML file created instead of actual feature names, feature names are displayed as x1,x2...,. Kindly help me how can I get actual feature name in PMML file?  PFB the data dictionary section from PMML and the code I have ran

**Data Dictionary Section from PMML:**
<DataDictionary>
		<DataField name="y" optype="categorical" dataType="integer">
			<Value value="0"/>
			<Value value="1"/>
			<Value value="2"/>
		</DataField>
		<DataField name="x2" optype="continuous" dataType="double"/>
		<DataField name="x3" optype="continuous" dataType="double"/>
		<DataField name="x4" optype="continuous" dataType="double"/>
	</DataDictionary>

**Code:**

from sklearn.datasets import load_iris
from sklearn2pmml import make_pmml_pipeline
from sklearn2pmml import sklearn2pmml

data = load_iris()
import pandas as pd

iris = pd.DataFrame(data.data , columns =  ['sepal_length','sepal_width','petal_length','petal_width'])
iris['target'] = data.target
iris = iris.sample(frac = 1)

from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test = train_test_split(iris,iris.target,test_size = 0.3, random_state = 42)

from sklearn.tree  import DecisionTreeClassifier

X_train.drop('target', axis = 1, inplace = True)
X_test.drop('target', axis = 1, inplace = True)

from sklearn.pipeline import Pipeline
from sklearn.impute import SimpleImputer
pipe = Pipeline([('imputer',SimpleImputer()),('model',DecisionTreeClassifier() )])

pipe.fit(X_train,y_train)

sklearn2pmml(make_pmml_pipeline(pipe), 'pipe_pmml.pmml')

**Comment by vruusmann** [comment]
> Kindly help me how can I get actual feature name in PMML file?

Before you come to complain about a project, please consult with its README file first:
https://github.com/jpmml/sklearn2pmml#usage

Right now you're using Scikit-Learn native `sklearn.pipeline.Pipeline` class, and you're passing Numpy arrays to its `Pipeline.fit(X, y)` method. These classes do not contain/collect feature information; the SkLearn2PMML package is informing you about this deficiency during conversion (please double check its console messages!), and is generating PMML documents that contain the maximum information that was given to it.

Now, if you do want to have feature names collected and stored automatically, you'd need to:
1. Switch from `sklearn.pipeline.Pipeline` to `sklearn2pmml.pipeline.PMMLPipeline`.
2. Switch from Numpy arrays to Pandas' data frames and series when invoking the `PMMLPipeline.fit(X, y)` method.

Alternatively, you may proceed with your existing setup, and create a properly configured `PMMLPipeline` object (off the fitted `Pipeline` object) using the `sklearn2pmml.make_pmml_pipeline(obj, active_fields, target_fields)` utility method.
